infra/examples-dev/aws-google-workspace/main.tf

```diff
@@ -0,0 +1,277 @@
+  connector_service_account_id = "psoxy-${each.key}"
+  display_name                 = "Psoxy Connector - ${each.value.display_name}${var.connector_display_name_suffix}"
+  apis_consumed                = each.value.apis_consumed
+  oauth_scopes_needed          = each.value.oauth_scopes_needed
+
+  depends_on = [
+    module.psoxy-aws,
+    google_project.psoxy-google-connectors
+  ]
+}
+
+module "google-workspace-connection-auth" {
+  for_each = module.worklytics_connector_specs.enabled_google_workspace_connectors
+
+  source = "../../modules/gcp-sa-auth-key-aws-secret"
+  # source = "git::https://github.com/worklytics/psoxy//infra/modules/gcp-sa-auth-key-aws-secret?ref=v0.4.3"
+
+  service_account_id = module.google-workspace-connection[each.key].service_account_id
+  secret_id          = "PSOXY_${replace(upper(each.key), "-", "_")}_SERVICE_ACCOUNT_KEY"
+}
+
+module "psoxy-google-workspace-connector" {
+  for_each = module.worklytics_connector_specs.enabled_google_workspace_connectors
+
+  source = "../../modules/aws-psoxy-rest"
+  # source = "git::https://github.com/worklytics/psoxy//infra/modules/aws-psoxy-rest?ref=v0.4.3"
+
+  function_name                         = "psoxy-${each.key}"
+  source_kind                           = each.key
+  path_to_function_zip                  = module.psoxy-aws.path_to_deployment_jar
+  function_zip_hash                     = module.psoxy-aws.deployment_package_hash
+  path_to_config                        = "${local.base_config_path}/${each.key}.yaml"
+  api_caller_role_arn                   = module.psoxy-aws.api_caller_role_arn
+  aws_assume_role_arn                   = var.aws_assume_role_arn
+  aws_account_id                        = var.aws_account_id
+  path_to_repo_root                     = var.psoxy_base_dir
+  example_api_calls                     = each.value.example_api_calls
+  example_api_calls_user_to_impersonate = each.value.example_api_calls_user_to_impersonate
+
+  parameters = [
+    module.psoxy-aws.salt_secret,
+    module.google-workspace-connection-auth[each.key].key_secret
+  ]
+}
+
+
+module "worklytics-psoxy-connection-google-workspace" {
+  for_each = module.worklytics_connector_specs.enabled_google_workspace_connectors
+
+  source = "../../modules/worklytics-psoxy-connection-aws"
+  # source = "git::https://github.com/worklytics/psoxy//infra/modules/worklytics-psoxy-connection-aws?ref=v0.4.3"
+
+  psoxy_endpoint_url = module.psoxy-google-workspace-connector[each.key].endpoint_url
+  display_name       = "${each.value.display_name} via Psoxy${var.connector_display_name_suffix}"
+  aws_region         = var.aws_region
+  aws_role_arn       = module.psoxy-aws.api_caller_role_arn
+}
+
+
+# BEGIN AUTH CONNECTORS
+
+locals {
+  enabled_oauth_long_access_connectors_todos = { for k, v in module.worklytics_connector_specs.enabled_oauth_long_access_connectors : k => v if v.external_token_todo != null }
+
+  secrets_to_create = distinct(flatten([
```

Comment:
new, now secrets to create are defined in `module.worklytics_connector_specs`
this flattens it in a format
[(CONN1, SECRET1), (CONN1, SECRET2), (CONN2, SECRET1)...]
to later be able to generate each pair.